fastplotlib/widgets/image.py

```diff
@@ -692,34 +647,36 @@ def _process_indices(
             dict in form of {dimension_index: slice_index}
             For example if an array has shape [1000, 30, 512, 512] corresponding to [t, z, x, y]:
                 To get the 100th timepoint and 3rd z-plane pass:
-                    {"t": 100, "z": 3}, or {0: 100, 1: 3}
+                    {"t": 100, "z": 3}
 
         Returns
         -------
         np.ndarray
             array-like, 2D slice
 
         """
-        indexer = [slice(None)] * self.ndim
+
+        data_ix = None
+        for i in range(len(self.data)):
+            if self.data[i] is array:
+                data_ix = i
+                break
 
         numerical_dims = list()
+        curr_ndim = self.data[data_ix].ndim
+        indexer = [slice(None)] * curr_ndim
+        curr_scrollable_format = SCROLLABLE_DIMS_ORDER[
```

Comment:
put a comment that this maps from n_scrollab_dims to one of: `"", "t", or "tz"`

generally explain these lines with line-by-line comments